File: Interviews/LeetCode/Daily-Problems/2022/November/Day-09/901.cpp
class StockSpanner {
   public:
    vector<vector<int>> st;
    StockSpanner() {
        st.clear();
    }

    int next(int price) {
        int crushed = 0;
        while (st.size() > 0 and st.back()[0] <= price) {
            crushed += 1 + st.back()[1];
            st.pop_back();
        }

        st.push_back({price, crushed});
        return 1 + crushed;
    }
};

File: Interviews/LeetCode/Daily-Problems/2022/November/Day-09/901.py
class StockSpanner:

    def __init__(self):
        self.st = []

    def next(self, price: int) -> int:
        crushed = 0
        while len(self.st) > 0 and self.st[-1][0] <= price:
            crushed += 1 + self.st[-1][1]
            self.st.pop()

        self.st.append([price, crushed])
        return 1 + crushed
